src/symmetric.jl

```diff
@@ -833,6 +833,12 @@ function svdvals!(A::RealHermSymComplexHerm)
     return sort!(vals, rev = true)
 end
 
+#computes U * Diagonal(abs2.(v)) * U'
+function _psd_spectral_product(v, U)
+    Uv = U * Diagonal(v)
+    return Uv * Uv'
```

Comment:
This is the reason for the speedup, right?
```suggestion
    return Uv * Uv' # often faster than generic matmul by calling BLAS.herk
```

Reply:
indeed